### PR TITLE
Edit and Remove Assessments (Frontend)

### PR DESCRIPTION
**Pull request by joaogreg**

### CODE REVIEW

File: backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/webservice/UpdateAssessmentWebServiceIT.groovy [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.assessment.AssessmentService
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.user.domain.User
+import org.springframework.http.MediaType
+import org.springframework.web.reactive.function.client.WebClient
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.SpockTest
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.utils.DateHandler
+
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class UpdateAssessmentWebServiceIT extends SpockTest {
```
Comment (ritosilva) [inline backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/webservice/UpdateAssessmentWebServiceIT.groovy:21]: write a test for a volunteer that did not write de assessment

File: frontend/tests/e2e/specs/assessments/edit_and_delete_assessment.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
+describe('Edit Assessment', () => {
```
Comment (ritosilva) [inline frontend/tests/e2e/specs/assessments/edit_and_delete_assessment.js:1]: include as tests of assessment.ts and remove this file

**Review by ritosilva — CHANGES_REQUESTED** [review]
Do the resquested changes.

File: backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/AssessmentController.java [hunk 3]
```diff
@@ -34,4 +34,17 @@ public AssessmentDto createAssessment(Principal principal, @PathVariable Integer
         int userId = ((AuthUser) ((Authentication) principal).getPrincipal()).getUser().getId();
         return assessmentService.createAssessment(userId, institutionId, assessmentDto);
     }
+
+    @DeleteMapping("/assessments/{assessmentId}")
+    @PreAuthorize("hasRole('ROLE_VOLUNTEER') and hasPermission(#assessmentId, 'ASSESSMENT.WRITER')")
+    public AssessmentDto deleteAssessment(Principal principal, @PathVariable int assessmentId) {
+        int userId = ((AuthUser) ((Authentication) principal).getPrincipal()).getUser().getId();
```
[the remark this answers is not in the document]
Comment (joaogreg) [reply backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/AssessmentController.java]: It's being used in frontend/src/services/RemoteServices.ts:654

[the remark this answers is not in the document]
Comment (ritosilva) [reply backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/AssessmentController.java:41]: "int userId" is not being used

[the remark this answers is not in the document]
Comment (ritosilva) [reply backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/AssessmentController.java:41]: I've changed it.

File: backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/domain/UpdateAssessmentMethodTest.groovy [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.user.domain.User
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.user.domain.Volunteer
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.utils.DateHandler
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.institution.domain.Institution
+import spock.lang.Unroll
+
+import java.time.LocalDateTime
+
+@DataJpaTest
+class UpdateAssessmentMethodTest extends SpockTest {
```
[the remark this answers is not in the document]
Comment (ritosilva) [reply backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/domain/UpdateAssessmentMethodTest.groovy:25]: OK

File: backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/service/CreateAssessmentServiceTest.groovy [hunk 5]
```diff
@@ -27,7 +27,7 @@ class CreateAssessmentServiceTest extends SpockTest {
         volunteer = authUserService.loginDemoVolunteerAuth().getUser()
 
         def activityDto = createActivityDto(ACTIVITY_NAME_1,ACTIVITY_REGION_1,1,ACTIVITY_DESCRIPTION_1,
-                TWO_DAYS_AGO,ONE_DAY_AGO,NOW,null)
+                THREE_DAYS_AGO,TWO_DAYS_AGO,ONE_DAY_AGO,null)
```
[the remark this answers is not in the document]
Comment (ritosilva) [reply backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/service/CreateAssessmentServiceTest.groovy:30]: Still open

File: backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/service/GetAssessmentsByInstitutionServiceTest.groovy [hunk 6]
```diff
@@ -21,7 +21,7 @@ class GetAssessmentsByInstitutionServiceTest extends SpockTest {
         otherInstitution = createInstitution(INSTITUTION_1_NAME, INSTITUTION_1_EMAIL, INSTITUTION_1_NIF)
 
         def activityDto = createActivityDto(ACTIVITY_NAME_1,ACTIVITY_REGION_1,1,ACTIVITY_DESCRIPTION_1,
-                TWO_DAYS_AGO, ONE_DAY_AGO,NOW,null)
+                THREE_DAYS_AGO, TWO_DAYS_AGO, ONE_DAY_AGO,null)
```
[the remark this answers is not in the document]
Comment (ritosilva) [reply backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/service/GetAssessmentsByInstitutionServiceTest.groovy:24]: Still open

File: backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/webservice/CreateAssessmentWebServiceIT.groovy [hunk 7]
```diff
@@ -31,7 +31,7 @@ class CreateAssessmentWebServiceIT extends SpockTest {
         def volunteer = authUserService.loginDemoVolunteerAuth().getUser()
 
         def activityDto = createActivityDto(ACTIVITY_NAME_1,ACTIVITY_REGION_1,1,ACTIVITY_DESCRIPTION_1,
-                TWO_DAYS_AGO,ONE_DAY_AGO,NOW,null)
+                THREE_DAYS_AGO, TWO_DAYS_AGO,ONE_DAY_AGO,null)
```
[the remark this answers is not in the document]
Comment (ritosilva) [reply backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/webservice/CreateAssessmentWebServiceIT.groovy:35]: Still open

File: backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/webservice/GetAssessmentsByInstitutionWebServiceIT.groovy [hunk 8]
```diff
@@ -31,10 +31,11 @@ class GetAssessmentsByInstitutionWebServiceIT extends SpockTest {
         institution = institutionService.getDemoInstitution()
 
         def activityDto = createActivityDto(ACTIVITY_NAME_1,ACTIVITY_REGION_1,1,ACTIVITY_DESCRIPTION_1,
-                TWO_DAYS_AGO, ONE_DAY_AGO,NOW,null)
+                THREE_DAYS_AGO,TWO_DAYS_AGO, ONE_DAY_AGO, null)
```
[the remark this answers is not in the document]
Comment (ritosilva) [reply backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/webservice/GetAssessmentsByInstitutionWebServiceIT.groovy:34]: Still open

File: backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/service/DeleteAssessmentServiceTest.groovy [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.exceptions.ErrorMessage
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.exceptions.HEException
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.auth.domain.AuthUser
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.user.domain.User
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.user.domain.Volunteer
+import spock.lang.Unroll
+
+
+@DataJpaTest
+class DeleteAssessmentServiceTest extends SpockTest {
```
[the remark this answers is not in the document]
Comment (ritosilva) [reply backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/service/DeleteAssessmentServiceTest.groovy:17]: Still open

File: backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/service/UpdateAssessmentServiceTest.groovy [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.exceptions.ErrorMessage
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.exceptions.HEException
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.auth.domain.AuthUser
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.user.domain.User
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.user.domain.Volunteer
+import spock.lang.Unroll
+import java.time.LocalDateTime
+
+@DataJpaTest
+class UpdateAssessmentServiceTest extends SpockTest {
```
[the remark this answers is not in the document]
Comment (ritosilva) [reply backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/service/UpdateAssessmentServiceTest.groovy:21]: Still open

File: backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/webservice/DeleteAssessmentWebServiceIT.groovy [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.institution.domain.Institution
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.user.domain.User
+import org.springframework.http.MediaType
+import org.springframework.web.reactive.function.client.WebClient
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.SpockTest
+import pt.ulisboa.tecnico.socialsoftware.humanaethica.utils.DateHandler
+
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class DeleteAssessmentWebServiceIT extends SpockTest {
```
[the remark this answers is not in the document]
Comment (ritosilva) [reply backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/webservice/DeleteAssessmentWebServiceIT.groovy:20]: Still open

[the remark this answers is not in the document]
Comment (ritosilva) [reply backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/webservice/DeleteAssessmentWebServiceIT.groovy:20]: Still open

File: backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/webservice/UpdateAssessmentWebServiceIT.groovy (hunk 1 above)
Comment (ritosilva) [reply backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/webservice/UpdateAssessmentWebServiceIT.groovy:21]: Still open

File: frontend/tests/e2e/specs/assessments/edit_and_delete_assessment.js (hunk 2 above)
Comment (ritosilva) [reply frontend/tests/e2e/specs/assessments/edit_and_delete_assessment.js:1]: Still open

**Comment by ritosilva** [comment]
Most of the issues were not addressed

File: backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/webservice/DeleteAssessmentWebServiceIT.groovy [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+
+        cleanup:
+        deleteAll()
+    }
+
+    def 'login as a volunteer that did not do the assessment'() {
+        given: 'another volunteer'
+        def otherInstitution = new Institution(INSTITUTION_1_NAME, INSTITUTION_1_EMAIL, INSTITUTION_1_NIF)
+        institutionRepository.save(otherInstitution)
+        def otherMember = createMember(USER_1_NAME,USER_1_USERNAME,USER_1_PASSWORD,USER_1_EMAIL, AuthUser.Type.NORMAL, otherInstitution, User.State.APPROVED)
```
Comment (ritosilva) [inline backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/webservice/DeleteAssessmentWebServiceIT.groovy:80]: It is not a volunteer, it is a member

File: backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/webservice/UpdateAssessmentWebServiceIT.groovy [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+
+        cleanup:
+        deleteAll()
+    }
+
+    def 'login as a volunteer that did not do the assessment'() {
+        given: 'another volunteer'
+        def otherInstitution = new Institution(INSTITUTION_1_NAME, INSTITUTION_1_EMAIL, INSTITUTION_1_NIF)
+        institutionRepository.save(otherInstitution)
+        def otherMember = createMember(USER_1_NAME,USER_1_USERNAME,USER_1_PASSWORD,USER_1_EMAIL, AuthUser.Type.NORMAL, otherInstitution, User.State.APPROVED)
```
Comment (ritosilva) [inline backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/assessment/webservice/UpdateAssessmentWebServiceIT.groovy:112]: It is not a volunteer, it is a member

**Review by ritosilva — CHANGES_REQUESTED** [review]
Access control tests need to be fixed

**Pull request merged**